array_test.go

```diff
@@ -108,6 +108,61 @@ func TestArray_Constructors(t *testing.T) {
 	})
 }
 
+func TestArray_Decode(t *testing.T) {
+	t.Run("Decode with slice of interface", func(t *testing.T) {
+		target := []interface{}{}
+		testValue := []interface{}{"Foo", 123.0}
+		reporter := newMockReporter(t)
+		arr := NewArray(reporter, testValue)
+		arr.Decode(&target)
+		arr.Equal(target)
+	})
+	t.Run("Decode with slice of struct", func(t *testing.T) {
+		reporter := newMockReporter(t)
+		type S struct {
+			Foo int `json:"foo"`
+		}
+		testValue := []interface{}{
+			map[string]interface{}{
+				"foo": 123,
+			},
+			map[string]interface{}{
+				"foo": 456,
+			},
+		}
+		arr := NewArray(reporter, testValue)
+		target := new([]S)
+		arr.Decode(target)
+		arr.Equal(target)
```

Comment:
Since we use mock reporter, arr.Equal check is not real (never marks test as failed).

Instead, we should use assert.Equal.

Also we should add arr.chain.assertNotFailed right after Decode call, to ensure that Decode succeeded.

Reply:
> Since we use mock reporter, arr.Equal check is not real (never marks test as failed).
   Instead, we should use assert.Equal. 

Yes, makes Sense. I actually tried this and wanted to clarify somethings..

So if we use assert.Equal , the types doesn't match because _value_ is map and _target_ is struct. So I tried comparing the values. I extracted the value.element(0).string().raw() data, which works fine. How do I extract the target data? Or is there another way?

Reply:
You could just create a variable of needed struct type snd value and compare target with that.